src/sockperf.cpp

```diff
@@ -1723,12 +1684,62 @@ static int proc_mode_server(int id, int argc, const char **argv) {
     return rc;
 }
 
+// Get resolved socket address from getaddrinfo() result. IPv6 is preferred.
+static void get_socket_address(struct addrinfo *result, struct sockaddr *addr, socklen_t &addrlen)
+{
+    for (addrinfo *rp = result; rp != NULL; rp = rp->ai_next) {
+        if (rp->ai_family == AF_INET6) {
+            std::memcpy(addr, rp->ai_addr, rp->ai_addrlen);
+            addrlen = rp->ai_addrlen;
+            return;
+        }
+    }
+    for (addrinfo *rp = result; rp != NULL; rp = rp->ai_next) {
+        std::memcpy(addr, rp->ai_addr, rp->ai_addrlen);
+        addrlen = rp->ai_addrlen;
+        return;
+    }
+    // This point is unreachable because getaddrinfo() returns EAI_NODATA if
+    // there are no network addresses defined for a host
+}
+
+//------------------------------------------------------------------------------
+static int resolve_sockaddr(const char *host, const char *port, int sock_type,
+        bool is_server_mode, sockaddr *addr, socklen_t &addr_len)
+{
+    struct addrinfo hints;
+
+    memset(&hints, 0, sizeof(hints));
+    // allow IPv4 or IPv6
+    hints.ai_family = AF_UNSPEC;
+    // return addresses only from configured address families
+    hints.ai_flags = AI_ADDRCONFIG;
+    // if host is NULL then server address will be 0.0.0.0/:: and client address will be 127.0.0.1/::1
```

Comment:
in current flow host can not be NULL.
and instead of introducing `is_server_mode ` host` variable can be used.
```
if (host == NULL) hints.ai_flags |= AI_PASSIVE;
```

I prefer not to change `parse_common_opt` prototype

Reply:
`host_str` in `parse_common_opt` can be `NULL` if `-i` option is not specified. This is rare but valid case.

Reply:
rewritten without this argument